reconcile/utils/openshift_resource.py

```diff
@@ -214,7 +214,11 @@ def api_version_mutation(val1, val2):
 
     @property
     def name(self):
-        return self.body["metadata"]["name"]
+        # PipelineRun name can be empty when creating
+        if self.kind == "PipelineRun" and "name" not in self.body["metadata"]:
+            return self.body["metadata"]["generateName"][:-1]
+        else:
+            return self.body["metadata"]["name"]
```

Comment:
since this logic is same as https://github.com/app-sre/qontract-reconcile/pull/4290/files#diff-006493679cf46d27e100feeb8475e6e29987fe85cca4858a1f05608b2d536dc8R166-R173, should extract an util function to be used for both to avoid later inconsistent

Reply:
PTAL https://github.com/app-sre/qontract-reconcile/pull/4290/commits/04491507f1e9e6feec020a0aa6446a4804d3fdf0